multipers/multiparameter_module_approximation/approximation.h

```diff
@@ -2306,53 +2300,62 @@ inline void Summand<value_type>::_add_birth(const filtration_type &birth) {
-  }
+  // // TODO:  Deprecate this old code
+  // if (death_corners_.empty()) {
+  //   death_corners_.push_back(death);
+  //   return;
+  // }
 
-  for (const auto &current_death : death_corners_) {
-    if (death <= current_death) {
-      return;
-    }
-  }
-  // this death value is useful, we can now remove useless other filtrations
-  for (auto &current_death : death_corners_) {
-    if (!current_death.empty() && (death >= current_death)) {
-      current_death.clear();
-    }
-  }
-  _clean(death_corners_);
-  death_corners_.push_back(death);
+  // for (const auto &current_death : death_corners_) {
+  //   if (death <= current_death) {
+  //     return;
+  //   }
+  // }
+  // // this death value is useful, we can now remove useless other filtrations
+  // for (auto &current_death : death_corners_) {
+  //   if (!current_death.empty() && (death >= current_death)) {
+  //     current_death.clear();
+  //   }
+  // }
+  // _clean(death_corners_);
+  // death_corners_.push_back(death);
 }
 
 template <typename value_type>
 inline value_type
 Summand<value_type>::_get_max_diagonal(const filtration_type &birth,
                                        const filtration_type &death,
                                        const Box<value_type> &box) const {
+  //assumes birth and death to be never NaN
   if constexpr (Debug::debug)
-    assert(birth.size() == death.size() && "Inputs must be of the same size !");
+    assert(!birth.is_finite || !death.is_finite || birth.size() == death.size() && "Inputs must be of the same size !");
+
   value_type s = inf;
   bool threshold_flag = !box.is_trivial();
-  if (threshold_flag)
-    for (unsigned int i = 0; i < birth.size(); i++) {
+  if (threshold_flag) {
+    unsigned int dim = std::max(birth.size(), box.dimension());
+    for (unsigned int i = 0; i < dim; ++i){
       value_type max_i =
           box.get_upper_corner().size() > i ? box.get_upper_corner()[i] : inf;
       value_type min_i = box.get_upper_corner().size() > i
-                             ? box.get_bottom_corner()[i]
-                             : negInf;
-      value_type t_death = std::min(death[i], max_i);
-      value_type t_birth = std::max(birth[i], min_i);
+                            ? box.get_lower_corner()[i]
+                            : negInf;
+      value_type t_death = death.is_plus_inf() ? max_i : (death.is_minus_inf() ? -inf : std::min(death[i], max_i));
+      value_type t_birth = birth.is_plus_inf() ? inf : (birth.is_minus_inf() ? min_i : std::max(birth[i], min_i));
+      s = std::min(s, t_death - t_birth);
+    }
+  } else {
+    unsigned int dim = std::max(birth.size(), death.size());
+    for (unsigned int i = 0; i < dim; i++){
+      //if they don't have the same size, then one of them has to (+/-)infinite.
+      value_type t_death = death.size() > i ? death[i] : death[0];  //assumes death is never empty
+      value_type t_birth = birth.size() > i ? birth[i] : birth[0];  //assumes birth is never empty
```

Comment:
I'm not a big fan of all of this.

Reply:
Can you be more precise? Is it more or less the same than before with just an extra tests for infinity values...

Reply:
This assumes inf is of the form {inf} + there is too many `..?..: .. ` imbriqué, which is not really readable. I haven't though about an alternative though